docs/rfcs/0007-export-direction.md

```diff
@@ -0,0 +1,221 @@
+```
+prometheus.remote_write "staging" {
+  receiver = [prometheus.scrape.demo1, prometheus.scrape.demo2] 
+  endpoint {
+    url = "http://mimir:9009/api/v1/push"
+  }
+}
+
+prometheus.scrape "demo1" {
+  targets = [{"__address__" = "127.0.0.1:12345"}]
+}
+
+prometheus.scrape "demo2" {
+  targets = [{"__address__" = "127.0.0.2:12345"}]
+}
+```
+
+### Example 2: Prometheus scrape converted to Otel format and sent to an Otel endpoint
+#### Example 2.1: How Flow works at the moment
+```
+prometheus.scrape "default" {
+  targets = [{"__address__"   = "127.0.0.1:12345"}]
+  forward_to = [otelcol.receiver.prometheus.default.receiver]
+}
+
+otelcol.receiver.prometheus "default" {
+  output {
+    metrics = [otelcol.exporter.otlp.default.input]
+  }
+}
+
+otelcol.exporter.otlp "default" {
+  client {
+    endpoint = env("OTLP_ENDPOINT")
+  }
+}
+```
+
+#### Example 2.2: How Flow could work if we do not use exports as inputs
+```
+prometheus.scrape "default" {
+    targets = [{"__address__"   = "127.0.0.1:12345"}]
+}
+
+otelcol.receiver.prometheus "default" {
+  receiver = prometheus.scrape.output
+}
+
+otelcol.exporter.otlp "default" {
+  // otelcol.receiver.prometheus.default.output will be of type 
+  // "otelcol.Consumer" instead of an "output" block
+  input = otelcol.receiver.prometheus.default.output
+  client {
+    endpoint = env("OTLP_ENDPOINT")
+  }
+}
+```
+
+## Potential approaches
+
+### Approach 1 - Use exports as both inputs and outputs (current status quo)
+With this approach the component's exports could be either its inputs or its outputs. The writer of each component uses their best judgement on whether having the exports as inputs or outputs would be most appropriate.
+
+#### Pros
+- Maximum flexibility.
+- Prometheus uses a pull model. Otel uses a push model. Supporting both models in Flow could make it easier to express Prometheus and Otel logic. 
+#### Cons
+- Lack of consistency between components can be confusing.
+- Having components in the same config file using different paradigms looks confusing.
+
+### Approach 2 - Use exports as inputs only
+With this approach the component's exports are its inputs. Each component will have to specify to which other components its output data is going to.
+
+#### Pros
+
+#### Cons
+- It may be awkward to modify components which are already configured, so that they send data to more downstream components.
+
+### Approach 3 - Use exports as outputs only
+With this approach the component's exports are its outputs. Other components will feed those outputs to their inputs. Each component will have to specify from which other components its input data is coming from.
+
+#### Pros
+- More intuitive. Newcomers to Flow will browse the documentation and look for inputs and outputs of each component. They will likely assume that the non-exported arguments are the component's inputs, whereas the exported ones are the component's outputs.
```

Comment:
I couldn't follow what this sentence meant. Let's try and use language consistent with the current Flow terminology and documentation.